gatsby-ssr.js

```diff
@@ -43,11 +40,37 @@ export const onRenderBody = ({ setHeadComponents }) => {
 
   if (storeConfig.analytics.gtmContainerId) {
     addPartytown = true
+    forward.push('dataLayer.push')
+
+    // The first script adds the GTM script to partytown. It is meant for when regular users
+    // are browsing the website, so that loading and executing it doesn't affect performance
+    //
+    // The second script is meant for GTM debugging. Since debugging GTM inside partytown still doesn't work,
+    // it is only executed when the url includes the gtm_debug query string.
+    //
+    // Since the query string isn't accessible during SSR, the decision of which script should be executed
+    // is bundled with the script, and that's why we need to include both. The script isn't GTM itself, but
+    // the code who will, after being executed, add the GTM script to the page.
     setHeadComponents([
-      <GoogleTagManager
+      <script
+        key="gtm.partytown"
+        type="text/partytown"
+        dangerouslySetInnerHTML={{
+          __html: googleTagManager({
+            containerId: storeConfig.analytics.gtmContainerId,
+            partytownScript: true,
+          }),
+        }}
+      />,
+      <script
         key="gtm"
-        containerId={storeConfig.analytics.gtmContainerId}
-        enablePartytown
+        type="text/javascript"
+        dangerouslySetInnerHTML={{
+          __html: googleTagManager({
+            containerId: storeConfig.analytics.gtmContainerId,
+            partytownScript: false,
+          }),
+        }}
```

Comment:
Have you ever thought about adding only 1 script?

I thought to put the snippet inside a template tag and in the browser, put the template content into a script tag, but idk if it will harm the performance.

Reply:
I've never thought of it, but I think it may harm performance and it makes the code more complex. In the end, I think the amount of code that'd be downloaded would be the same, so I don't feel like implementing it.